### ISSUE TITLE
Timeout/network errors not throwing when network is down

### ISSUE THREAD

**Issue by wemersonrv** [axios/axios#3201]
#### Describe the issue
Testing timeout, or network errors. So trying to download a large file and after starts, shutdown my network. But no error is fired.

#### Example Code
```js
const axios = require('axios').default
const url = 'https://github.com/Genymobile/scrcpy/releases/download/v1.16/scrcpy-win64-v1.16.zip'

const response = await axios.get(url, {
   timeout: 1000,
   responseType: 'arraybuffer',
})

if(response.status === 200){
   console.log('File download is complete')
}
```

#### Expected behavior, if applicable
Throwing some network or timeout errors

#### Environment
 - Axios Version [e.g. 0.18.0]: ^0.19.2
 - Adapter [e.g. XHR/HTTP]: HTTP
 - Browser [e.g. Chrome, Safari]: node.js
 - Browser Version [e.g. 22]: node.js
 - Node.js Version [e.g. 13.0.1] v12.5.0
 - OS: [e.g. iOS 12.1.0, OSX 10.13.4] Windows 10
 - Additional Library Versions [e.g. React 16.7, React Native 0.58.0] [file-system](https://www.npmjs.com/package/file-system)

#### Additional context/Screenshots
Just need to understand how to intercept these kind of errors, because nothing happens

**Comment by agnjunio** [comment]
This seems to be intentional, as mentioned before in https://github.com/axios/axios/issues/647

Additional information in stack overflow: https://stackoverflow.com/questions/36690451/timeout-feature-in-the-axios-library-is-not-working/54573024

Although it seems to be a recurrent issue, maybe we should add some option to include network timeouts in this timeout setting.

**Comment by wemersonrv** [comment]
Understand. 

I think if it had options to catch network issues would be great!

**Comment by NoxHarmonium** [comment]
FWIW I ended up using an agent to set the socket time-out and it seems to be working well for me now.

You can demonstrate the different time-out behaviours in the node REPL

```javascript
const Axios = require('axios');
const http = require('http');
const agentWithoutTimeout = new http.Agent();
const agentWithTimeout = new http.Agent({ timeout: 2000 });
const instance1 = Axios.create({ timeout: 5000, httpAgent: agentWithoutTimeout });
const instance2 = Axios.create({ timeout: 5000, httpAgent: agentWithTimeout });
// http://www.google.com:81/ will always timeout because it drops SYN packets (see https://stackoverflow.com/a/100859/1153203)
instance1.get("http://www.google.com:81/").catch(e => { if (e.code === "ECONNABORTED" ) { console.log("instance 1 socket timed out") } else { console.log(e) } });
instance2.get("http://www.google.com:81/").catch(e => { if (e.code === "ECONNABORTED" ) { console.log("instance 2 socket timed out") } else { console.log(e) } });
 ```

You will see "instance 2 timed out" after two seconds but instance1 won't time out

The two timeouts are also independent, the agent timeout can be lower than than Axios timeout.

If the request is cancelled due to the socket timeout set on the HTTP agent, Axios will throw an error reporting that the timeout occurred with the value specified in the Axios config. For example, you'll see on the above example, if you log the error, it will report that the timeout was 5000ms when it actually timed out after 2000ms. This behaviour does seem like me to be a bug rather than just a missing feature.

**Comment by vmatyusGitHub** [comment]
I am also facing this problem, we have a change feed service, that notifies the client application about any change on the server.
When the network connection is lost for a longer period of time (more than 10min) the request connection seems to be stable -no connection closed message received.
After when I set back the network connection and I make any changes on the server the client is not notified anymore about the change.

So somehow the connection is lost between the client and the server.
I would like also intercept these type of errors. At least print out something on the client side.

**Comment by weareu** [comment]
@vmatyusGitHub We have the same issue to a couch feed. Fixed it by implementing as described in #647

let source = CancelToken.source();
    setTimeout(() => {
        source.cancel();
}, 300);
return axios.get(ip + '/url', {cancelToken: source.token}).then((result) => {
 // code here
});

**Comment by jasonsaayman** [comment]
Closing please see #647
